client/src/redux/reducers/apps.reducers.js

```diff
@@ -7,12 +7,34 @@ const initialState = {
   defaultTab: ''
 };
 
+function unpackAppIcons(tabs) {
+  const appIcons = {};
+  tabs.forEach(tab => {
+    tab.apps.forEach(appEntry => {
+      if ('icon' in appEntry && appEntry.icon.length > 0) {
```

Comment:
If `appEntry.icon` turns out to be null or undefined the `appEntry.icon.length > 0` will throw an error I think. Should we use something like `if ('icon' in appEntry && appEntry.icon) {` instead?